cmd/dump/src/lib.rs

```diff
@@ -616,36 +693,154 @@ fn read_dump_header(
     read_dump_at(hubris, core, context, funcs, 0, |_| false)
 }
 
+fn parse_dump_header(
+    area: usize,
+    buf: &[u8],
+) -> Result<(DumpAreaHeader, Option<DumpTask>)> {
+    let header = DumpAreaHeader::read_from_prefix(buf)
+        .ok_or_else(|| anyhow!("failed to parse dump area {area}"))?;
+
+    if header.magic != humpty::DUMP_MAGIC {
+        bail!("bad magic at in dump area {area}: {header:x?}");
+    }
+
+    //
+    // This is a little sleazy (or maybe even a lot sleazy?):  we know that if
+    // we have task dump here, the number of segments is sufficiently low to
+    // assure that we will have also slurped our task information -- and we
+    // know that our task information will immediately follow our segment
+    // headers.  Gone fishin'...
+    //
+    let size = size_of::<DumpAreaHeader>();
+    let toffs =
+        size + header.nsegments as usize * size_of::<DumpSegmentHeader>();
+
+    let task = if toffs < buf.len() {
+        if let Some(DumpSegment::Task(t)) = DumpSegment::from(&buf[toffs..]) {
+            Some(t)
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+
+    Ok((header, task))
+}
+
 fn read_dump_headers(
     hubris: &HubrisArchive,
     core: &mut dyn Core,
     context: &mut HiffyContext,
     funcs: &HiffyFunctions,
-) -> Result<Vec<DumpAreaHeader>> {
+    raw: bool,
+) -> Result<Vec<(DumpAreaHeader, Option<DumpTask>)>> {
+    let rsize = 256;
+    let chunksize = (context.rdata_size() / rsize) - 1;
+    let op = hubris.get_idol_command("DumpAgent.read_dump")?;
+    let mut base = 0;
     let mut rval = vec![];
-    let mut done = false;
-    let mut ndx = 0;
 
-    while !done {
-        let header =
-            read_dump_at(hubris, core, context, funcs, ndx, |_| false)?;
+    //
+    // Iterate over dump area headers in chunksize units.  Once we've hit an
```

Comment:
Add a note that this only works because the area header is less than `rsize` here.